src/sharding/ShardClientUtil.js

```diff
@@ -224,6 +227,18 @@ class ShardClientUtil {
     }
     return this._singleton;
   }
+
+  /**
+   * Get the shard Id for a given guild Id.
+   * @param {Snowflake} guildId Snowflake guild Id to get shard Id for
+   * @param {number} shardCount Number of shards
+   * @returns {number}
+   */
+  static shardIdForGuildId(guildId, shardCount) {
+    // This performs (guild_id >> 22) % num_shards, but with a string Snowflake
+    const timestamp = Snowflake.deconstruct(guildId).timestamp;
+    return (timestamp - Snowflake.epoch) % shardCount;
+  }
```

Comment:
@pizzafox @advaith1 lmk if this implementation makes sense based on what y'all had done

Reply:
There should be validation here. On mobile, will post what I do in a second.

Reply:
```js
static shardIdForGuildId(guildId, shardCount) {
  // This performs (guild_id >> 22) % num_shards, but with a string Snowflake
  const timestamp = Snowflake.deconstruct(guildId).timestamp;
  
  const shard = (timestamp - Snowflake.epoch) % shardCount;
  
  if (calculated >= 0 && calculated < shardCount) {
    return shard;
  }

  throw new Error(`shardCount is incorrect or guildId ${guildID} is invalid, calculated ${calculated}`);
}
```
Something like this, but with the funny errors file Discord.js uses.

Reply:
Huh? All snowflakes are bigger than MAX_SAFE_INTEGER. Only reason you could not notice is because your older ID ends with 0

Reply:
`Snowflake#deconstruct` uses strings for the full snowflake + binary representation, only casting to integers to return the resultant object.

Reply:
I'm probably misremembering what specifically happened, but it was due to some number being too precise for a double. Wrapping everything in `BigInt` fixed it.

Reply:
It's fine to use this code, we require Node.js v12, and according to [node.green](https://node.green#ES2020-features-BigInt), it's available since v10.8.0.
```javascript
const shard = Number(BigInt(guildId) >> 22n) % shardCount;
```

Reply:
*keep in mind you will need to change the `ecmaVersion` to `2020` in `.eslintrc.json` if you choose to use the `BigInt(guildId)` approach*

Reply:
👍 Using the BigInt option seems like a nice, clean option. It is the first use of it in the codebase though, so yeah, ecma needs bumping :)

Reply:
> _keep in mind you will need to change the `ecmaVersion` to `2020` in `.eslintrc.json` if you choose to use the `BigInt(guildId)` approach_

It's fine, Node.js v12 runs ECMAScript 2020, which also includes `BigInt`. Besides, this wouldn't be a breaking change at all, just something that should have been done a long time ago.

Reply:
yeah, just noting that it needs to be done